src/views/signup.jsx

```diff
@@ -1,7 +1,185 @@
-import React from "react";
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  createUserWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "../../firebase_config";
 
-const signup = () => {
-  return <div>signup</div>;
-};
+function Signup() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
 
-export default signup;
+  const provider = new GoogleAuthProvider();
+  const navigate = useNavigate();
+
+  const handleSignUp = async (e) => {
+    e.preventDefault();
+
+    if (password !== confirmPassword) {
+      alert("password don't match");
+      return;
+    }
+
+    createUserWithEmailAndPassword(auth, email, password)
+      .then(() => {
+        alert("sign up successfully");
+      })
+      .catch((error) => {
+        if (error.code === "auth/email-already-in-use") {
+          alert("email address is already in use");
+        }
+        if (error.code === "auth/invalid-email") {
+          alert("email address is invalid");
+        }
+        console.log(error);
+      });
+  };
+
+  const handleGoogleSignUp = async () => {
+    signInWithPopup(auth, provider)
+      .then(() => {
+        navigate("/dashboard");
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  return (
+    <div className="flex min-h-full flex-1 flex-col justify-center py-12 sm:px-6 lg:px-8">
+      <div className="sm:mx-auto sm:w-full sm:max-w-md">
+        <Link to="/">
+          <img
+            className="mx-auto h-28 w-auto"
+            src="https://res.cloudinary.com/yilin1234/image/upload/v1692498189/f__2_-removebg-preview_xxhyv5.png"
```

Comment:
we can store the image urls in a constants file since they are reused in other parts of the application.